### ISSUE TITLE
Diacritics from external keyboard not working

### ISSUE THREAD

**Issue by voidbert** [gujjwal00/avnc#14]
**Version**: The latest from F-Droid as of now (1.2.0) on Android 10

**How to recreate from scratch:**
- Run Ubuntu chrooted through [Termux](https://f-droid.org/en/packages/com.termux/) (using [AnLinux](https://f-droid.org/en/packages/exa.lnx.a/) for the installation).

- Also use AnLinux to install XFCE. Run `./start-ubuntu.sh` and `vncserver-start`. Connect AVNC to 127.0.0.1:1.

- In XFCE's Keyboard settings, the "Use system defaults" option is ticked.

**Other (maybe useful) info:**

- If this XFCE option weren't ticked, there would be no Keyboard model options (like pc104 and others)
- `setxkbmap pt` prints out `XKB extension not present on :1.0`, what I expected after reading the AVNC code's documentation

**Problem:**

When using the onscreen keyboard (brought up by swiping from the left and pressing the keyboard button), I can input all characters, including the ones with diacritics.

When using a Bluetooth keyboard, the layout on my keyboard (PT-pt) and the text input match perfectly. The problem comes up when I try to input an accented character. I press the acute accent key (and release it), then I press the key of the letter (á -> ´ + a). Only the letter is registered.

I used an online keyboard tester tool (https://www.keyboardtester.com/tester.html) and, when I clicked on the accent, no key presses at all were registered. When I pressed the letter, only it was registered.

**What I tried to do to fix the problem:**

Came complaining to the issues tab (just kidding). I tried reading the source code to see if, by any chance, I could quickly fix the problem or at least know what was happening. By the way, I have to congratulate you on the excellent documentation! The issue is probably in [KeyHandler.kt](https://github.com/gujjwal00/avnc/blob/master/app/src/main/java/com/gaurav/avnc/ui/vnc/KeyHandler.kt).

My best "Android" development experience has been Cordova, so take my guess of what is happening with a big pile of salt: When I press ´ and the the letter, two events are triggered, of which only the second is being considered valid and is emitted. Though, when I use the onscreen keyboard, only one event is triggered (á).

**Other:**

There can be three events that result in a single chacater: Shift + ´ + a = à (in PT-pt layout). There might be other languages where there can be more events, even letters with two or more diacritics. These are notes to make sure any eventual fix of this problem in the code doesn't just fix this specific issue with two events, when there can be more.

**Comment by gujjwal00** [comment]
Thanks for this very detailed report! I will look into it (although you have already done most the work 😀).

**Comment by gujjwal00** [comment]
It's taking more time than expected. Can you please tell me which VNC server & version you are using? Different servers seems to handle diacritics differently (TightVNC on Windows seems to work but Vino on Ubuntu is failing with AVNC).

**Comment by voidbert** [comment]
`apt list --installed | grep vnc` returns `tightvncserver/focal.now 1.3.10-0ubuntu5 arm64 [installed]`

**Comment by gujjwal00** [comment]
I have got it working with TightVNC atleast (Windows + Linux). I am uploading a test APK here. Please check if it handles accents correctly for you.

[app-debug.zip](https://github.com/gujjwal00/avnc/files/7495335/app-debug.zip)

Some things you can try:
- Simple accents
- With capital letters
- Multiple accents on same character
- Invalid combinations of accents & letters 
- Backspace after accent

**Comment by voidbert** [comment]
Hey! Sorry for delay. I've been quite busy lately. Diacritics are working very well for text editing. I can finally replace Collabora Office with full Libre Office through Termux and, somehow, it performs better. Thanks for all your help!


I tested everything you mentioned and all features seem to be working. Though great for editing text, there's a little detail missing. In a usual input method, if you press the accent key twice, it inputs that accent as a single character. Though, when you press the accent key twice in AVNC, it just cancels all accents. When I try to input a character next, it isn't accented (normal behavior). Though, the accent character (´) isn't inputted before.

**Comment by gujjwal00** [comment]
>  In a usual input method, if you press the accent key twice, it inputs that accent as a single character. Though, when you press the accent key twice in AVNC, it just cancels all accents

Yeah, I missed this case. I will fix this, and include diacritics support in next release (probably next week).

**Comment by voidbert** [comment]
Thanks for your help!
